packages/expo/src/hooks/useSSO.ts

```diff
@@ -0,0 +1,106 @@
+import { useSignIn, useSignUp } from '@clerk/clerk-react';
+import type { EnterpriseSSOStrategy, OAuthStrategy, SetActive, SignInResource, SignUpResource } from '@clerk/types';
+import * as AuthSession from 'expo-auth-session';
+import * as WebBrowser from 'expo-web-browser';
+
+import { errorThrower } from '../utils/errors';
+
+export type UseSSOParams = {
+  strategy: OAuthStrategy | EnterpriseSSOStrategy;
+  unsafeMetadata?: SignUpUnsafeMetadata;
+  redirectUrl?: string;
+};
+
+export type StartSSOParams = {
+  identifier?: string;
+  unsafeMetadata?: SignUpUnsafeMetadata;
+  redirectUrl?: string;
+};
+
+export type StartSSOFlowReturnType = {
+  /**
+   * Session ID created upon sign-in completion, or null if incomplete.
+   * If incomplete, use signIn or signUp for next steps like MFA.
+   */
+  createdSessionId: string | null;
+  setActive?: SetActive;
+  signIn?: SignInResource;
+  signUp?: SignUpResource;
+  authSessionResult?: WebBrowser.WebBrowserAuthSessionResult;
+};
+
+export function useSSO(useSSOParams: UseSSOParams) {
+  const { signIn, setActive, isLoaded: isSignInLoaded } = useSignIn();
+  const { signUp, isLoaded: isSignUpLoaded } = useSignUp();
+
+  async function startFlow(startSSOFlowParams: StartSSOParams = {}): Promise<StartSSOFlowReturnType> {
+    if (!isSignInLoaded || !isSignUpLoaded) {
+      return {
+        createdSessionId: null,
+        signIn,
+        signUp,
+        setActive,
+      };
+    }
+
+    let createdSessionId = signIn.createdSessionId;
+
+    const redirectUrl =
+      startSSOFlowParams?.redirectUrl ||
+      useSSOParams.redirectUrl ||
+      AuthSession.makeRedirectUri({
+        path: 'sso-native-callback',
+      });
+
+    await signIn.create({
+      strategy: useSSOParams.strategy,
+      redirectUrl,
+      identifier: startSSOFlowParams.identifier,
+    });
+
+    const { externalVerificationRedirectURL } = signIn.firstFactorVerification;
+    if (!externalVerificationRedirectURL) {
+      return errorThrower.throw(
+        'Missing external verification redirect URL for SSO flow. This indicates an API issue - please contact support for assistance.',
+      );
+    }
+
+    const authSessionResult = await WebBrowser.openAuthSessionAsync(
+      externalVerificationRedirectURL.toString(),
+      redirectUrl,
```

Comment:
Something to consider, is that our OAuth callback or /ACS endpoint already resolve with a 301 status code the deep link, therefore this second argument is not really needed